src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -307,11 +360,88 @@ const getFormData = async (formUuid, dispatch) => {
             type: 'FORM_STEPS_LOADED',
             payload: formStepsData,
         });
+        formatRuleComponentChoices(formStepsData, dispatch, 'FORMATTED_RULES_COMPONENTS_CHOICES');
     } catch (e) { // TODO: convert to ErrorBoundary
         dispatch({type: 'SET_FETCH_ERRORS', payload: {loadingErrors: e.message}});
     }
 };
 
+
+const getFormDefinitions = async (dispatch) => {
+    try {
+        const response = await get(FORM_DEFINITIONS_ENDPOINT);
+        dispatch({
+            type: 'FORM_DEFINITIONS_LOADED',
+            payload: response.data.results,
+        });
+    } catch (e) {
+        dispatch({type: 'SET_FETCH_ERRORS', payload: {loadingErrors: e.message}});
+    }
+};
+
+const getRegistrationBackends = async (dispatch) => {
+    try {
+        const response = await get(REGISTRATION_BACKENDS_ENDPOINT);
+        dispatch({
+            type: 'REGISTRATION_BACKENDS_LOADED',
+            payload: response.data
+        });
+    } catch (e) {
+        dispatch({type: 'SET_FETCH_ERRORS', payload: {loadingErrors: e.message}});
+    }
+};
+
+const getRegistrationBackendOptions = async (dispatch) => {
+    try {
+        const response = await get(REGISTRATION_BACKEND_OPTIONS_ENDPOINT);
+        dispatch({
+            type: 'REGISTRATION_BACKEND_OPTION_FORM_SCHEMAS_LOADED',
+            payload: response.data
+        });
+    } catch (e) {
+        dispatch({type: 'SET_FETCH_ERRORS', payload: {loadingErrors: e.message}});
+    }
+};
+
+const getAuthPlugins = async (dispatch) => {
+    try {
+        const response = await get(AUTH_PLUGINS_ENDPOINT);
+        dispatch({
+            type: 'AUTH_PLUGINS_LOADED',
+            payload: response.data
+        });
+    } catch (e) {
+        dispatch({type: 'SET_FETCH_ERRORS', payload: {loadingErrors: e.message}});
+    }
+};
+
+const getPrefillPlugins = async (dispatch) => {
+    try {
+        const response = await get(PREFILL_PLUGINS_ENDPOINT);
+        dispatch({
+            type: 'PREFILL_PLUGINS_LOADED',
+            payload: response.data
+        });
+    } catch (e) {
+        dispatch({type: 'SET_FETCH_ERRORS', payload: {loadingErrors: e.message}});
+    }
+};
```

Comment:
This was refactored in the master branch, please check the current `useAsync` implementation with `loadPlugins`.